test/jdk/java/foreign/TestByteBuffer.java

```diff
@@ -390,7 +390,8 @@ public void testBufferOnClosedScope() {
         try (MemorySegment segment = MemorySegment.allocateNative(bytes)) {
             leaked = segment;
         }
-        leaked.asByteBuffer();
+        ByteBuffer byteBuffer = leaked.asByteBuffer(); // ok
+        byteBuffer.get(); // should throw
     }
```

Comment:
I am a little confused with this change. What can we do with the returned ByteBuffer? Should we check the liveness  in asByteBuffer()?

Reply:
This change is just a merge change from the fix of JDK-8240874. The goal of that change is to determine a failure caused by liveness. Before this change the code was (redundantly) treating asByteBuffer as an operation which could only be performed on live segments. I've rectified that, but that means this test no longer threw the expected exception, so I added another call which was triggering the failure (get).